testFramework/com/twitter/intellij/pants/testFramework/PantsIntegrationTestCase.java

```diff
@@ -493,6 +493,17 @@ protected ScalaTestRunConfiguration generateScalaRunConfiguration(String moduleN
     return runConfiguration;
   }
 
+  protected void gitCleanExampleDir() throws ExecutionException {
+    // Git reset .cache/pants dir
+    cmd("git", "reset", "--hard");
+    // Only the files under examples are going to be modified.
+    // Hence issue `git clean -fdx` under examples, so pants does not
+    // have to bootstrap again.
+    File exampleDir = new File(getProjectFolder(), "examples");
```

Comment:
I know it's an extraction of existing code but I found this comment a little unclear. How about

Tests only modify sources under examples, so `git clean -fdx` just that directory. This avoids having to bootstrap pants again.